scripts/expressions.py

```diff
@@ -0,0 +1,126 @@
+import re
+from typing import Optional
+
+
+class Expression:
+    """
+    Helper class used to break down (and validate) arithmetic expressions
+    Examples of usage:
+    >>> Expression('5')
+    5
+    >>> Expression('0xff')
+    255
+    >>> Expression('037')
+    31
+    >>> Expression('n')
+    'n'
+
+    >>> e = Expression('5*n')
+    >>> e.left
+    5
+    >>> e.op
+    '*'
+    >>> e.right
+    'n'
+
+    >>> e = Expression('a+b-c')
+    >>> e.left
+    'a'
+    >>> e.op
+    '+'
+    >>> e.right
+    Expression('b-c')
+    """
+
+    operators = re.compile(r'[-+*/]')
+
+    def __init__(self, expr: str = None):
+        left, self.op, right = _split(expr)
+        self.left = Expression(left)
+        self.right = Expression(right)
+
+    def __new__(cls, expr: str = None):
```

Comment:
Explain why `__new__` is needed

Reply:
Size is alway spassed to the constructor and `__new__` returns an `int`, a `str`, `None`, or an `Expression`. Not exactly a common practice, but I've seen worse uses for `__new__`

Reply:
Then add a short comment in the code explaining that practice.